binders/kafka-binder/spring-cloud-stream-binder-kafka-streams/src/main/java/org/springframework/cloud/stream/binder/kafka/streams/SerdeResolverUtils.java

```diff
@@ -48,124 +47,120 @@ abstract class SerdeResolverUtils {
 
 	private static final Log LOG = LogFactory.getLog(SerdeResolverUtils.class);
 
+	/** Classnames of the standard built-in Serdes supported in {@link Serdes#serdeFrom(Class)} */
+	private static final Set<String> STANDARD_SERDE_CLASSNAMES = Set.of(
+		Serdes.String().getClass().getName(),
+		Serdes.Short().getClass().getName(),
+		Serdes.Integer().getClass().getName(),
+		Serdes.Long().getClass().getName(),
+		Serdes.Float().getClass().getName(),
+		Serdes.Double().getClass().getName(),
+		Serdes.ByteArray().getClass().getName(),
+		Serdes.ByteBuffer().getClass().getName(),
+		Serdes.Bytes().getClass().getName(),
+		Serdes.UUID().getClass().getName());
```

Comment:
Why names? Why just classes not enough?
Looks like you have the target class as well to check its existence in this set.
Otherwise you do extra step to take the name of that class.

Reply:
Yep, it is intentional and the reason is fear of holding on to a class reference vs. a reference to a string. Probably paranoia though.